lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -397,6 +402,17 @@ swift::matchWitness(
   } else if (isa<ConstructorDecl>(witness)) {
     decomposeFunctionType = true;
     ignoreReturnType = true;
+  } else if (isa<EnumElementDecl>(witness)) {
+    auto enumCase = cast<EnumElementDecl>(witness);
+    if (enumCase->hasAssociatedValues())
+      return RequirementMatch(witness, MatchKind::EnumCaseWithAssociatedValues);
+    auto var = dyn_cast<VarDecl>(req);
+    if (!var)
+      return RequirementMatch(witness, MatchKind::KindConflict);
+    if (!var->isStatic())
+      return RequirementMatch(witness, MatchKind::StaticNonStaticConflict);
+    if (var->getParsedAccessor(AccessorKind::Set))
```

Comment:
I think var->isMutable() would be a better query for this.

Reply:
There is no `isMutable()` method on `VarDecl` unfortunately.